Merge back 'chore_release-pd-8.5.0' into 'edge'

ddcc4 opened the pull request:
This propagates the changes we made today in `chore_release-8.5.0`.

ecormany committed:
docs(api): resolve TODO for 2.23 changes to InstrumentContext.name (#18702)

# Overview

See here's the thing about TODOs: you actually have _to do_ them.

thx @ddcc4 

## Test Plan and Hands on Testing

there's [a
sandbox](http://sandbox.docs.opentrons.com/pipette-name-todo-finally/v2/new_protocol_api.html#opentrons.protocol_api.InstrumentContext.name)

## Changelog

uncommented the thing we were supposed to a while ago!

## Risk assessment

negative

sfoster1 committed:
fix(hardware-testing): fix hanging gravi analysis (#18685)

You can't really use click.test's isolated environments because it needs
to be able to import hardware testing. It also needs to clean up its
recording thread on failure in addition to on success, and finally doing
these thigns reveals some failures we'll also have to fix.

---------

Co-authored-by: Ryan Howard <ryan.howard@opentrons.com>

mjhuff committed:
refactor(app): round pipetting command params to 2 sig figs (#18708)

Closes RQA-3937

Rounds all pipetting command params to two significant figures.

mjhuff committed:
refactor(shared-data): update lid def compatibility (#18709)

sanni-t committed:
fix(api): require LC transfers to use API v2.24 (#18712)

Closes AUTH-2004

# Overview

Updates the liquid classes based transfer, consolidate and distribute
methods to require 2.24 instead of 2.23. Details in ticket above for why
we need to do this.

## Risk assessment

Not really. We had previously indicated in api source code that liquid
classes work is in-progress and subject to change. And these methods are
not available in public docs until 8.5.0.

ddcc4 committed:
Merge back 'chore_release-8.5.0' into 'chore_release-pd-8.5.0' (#18714)

sfoster1 commented:
@ddcc4 for the `.flake8` conflict, take the part where it adds the venv ignores and delete the line-length restriction (disabling `E203` below turns off line length checking entirely in favor of running `black --check`)

sanni-t committed:
fix(shared-data): update lc display names (#18711)

Closes AUTH-2003

# Overview

Updates the display names to correct ones shown in design proposals.

## Risk assessment

None.

---------

Co-authored-by: Jethary <jethary@opentrons.com>

jerader committed:
refactor(protocol-designer): disallow lids in PD 8.5.0 (#18719)

closes RQA-4302

koji committed:
fix(workflow): Fix sentry build issue on ci (#18723)

*fix(workflow): Fix sentry build issue on ci

jerader reviewed (approved):
lgtm! waiting on this mergeback to merge in so i cut alpha.1 for labware library 😄 no rush though, anytime this morning is fine.

ncdiehl11 committed:
feat(step-generation): ensure retract position is safe for air gap (#18730)

This PR introduces logic to check whether or not the retract position is
safe for post-aspirate or -dispense air gaps. If the retract position is
2mm or more above the top of the well (deemed to be a "safe" position
for air gap), we can move to the retract position for air gap.
Otherwise, we move to the safe position above the well, and air gap
there.

jbleon95 committed:
fix(api): fix chained transfers with keep_last_tip set to True and return_tip (#18715)

Fixes bugs related to tracking picked up or returned tips between liquid class transfers

sanni-t committed:
fix(api): error raising when there's no innergeometry for a well (#18728)

Closes RQA-4304

# Overview

The check that verifies that a submerge start location or retract end
location is not inside the liquid in well relies on knowing the height
of the liquid- either by calculating it based on liquid volume and well
inner geometry or by probing into the well.
We already skip the check if we don't know the liquid volume or haven't
probed the well yet, opting to simply log the decision instead. We
should also skip the check if a well's innergeometry isn't present
either otherwise the check will raise errors for any labware that
doesn't have inner geometry defined, like the `smc_384_read_plate`, or
if it's a custom labware.

So this PR makes it so that the check excepts on
`IncompleteLabwareDefinitionError` as well.

## Risk assessment

Low. Only skips a check that would have errored without being useful.

ddcc4 committed:
test(components): fix CommandText.test.tsx after we changed liquid class displayNames (#18738)

# Overview

This fixes a broken test after PR #18711, which changed the names of our
liquid classes (AUTH-2003).

Specifically, we renamed `Volatile` to `Volatile (80% ethanol)`.

## Test Plan and Hands on Testing

Ran test locally. Will also see if CI tests pass.

## Risk assessment
Low, test only.

ddcc4 committed:
Merge back 'chore_release-8.5.0' into 'chore_release-pd-8.5.0'

Had to fix conflict in `protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/__tests__/LiquidClassesStepTools.test.tsx`.

ncdiehl11 committed:
fix(protocol-designer): fix max conditioning volume (#18737)

This PR fixes the logic for determining the max conditioning volume
allowed in a multiDispense transfer. We should always leave room enough
in the tip to accommodate a minimum of 2 destinations-worth of volume.
